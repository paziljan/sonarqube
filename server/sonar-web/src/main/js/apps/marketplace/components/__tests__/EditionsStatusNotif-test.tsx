/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import EditionsStatusNotif from '../EditionsStatusNotif';

it('should display an in progress notif', () => {
  const wrapper = shallow(
    <EditionsStatusNotif editionStatus={{ installationStatus: 'AUTOMATIC_IN_PROGRESS' }} />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should display an error notification', () => {
  const wrapper = shallow(
    <EditionsStatusNotif editionStatus={{ installationStatus: 'AUTOMATIC_FAILURE' }} />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should display a ready notification', () => {
  const wrapper = shallow(
    <EditionsStatusNotif editionStatus={{ installationStatus: 'AUTOMATIC_READY' }} />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should display install errors', () => {
  const wrapper = shallow(
    <EditionsStatusNotif
      editionStatus={{ installationStatus: 'AUTOMATIC_IN_PROGRESS', installError: 'Foo error' }}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
