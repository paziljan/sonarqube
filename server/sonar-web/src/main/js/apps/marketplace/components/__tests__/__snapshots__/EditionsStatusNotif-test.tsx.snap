// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a ready notification 1`] = `<div />`;

exports[`should display an error notification 1`] = `<div />`;

exports[`should display an in progress notif 1`] = `<div />`;

exports[`should display install errors 1`] = `
<div>
  <div
    className="alert alert-danger"
  >
    Foo error
  </div>
</div>
`;
